src/modules/verifiable-credentials/verifiable-credentials-base.service.ts

```diff
@@ -702,4 +697,22 @@ export abstract class VerifiableCredentialsServiceBase {
       return null;
     }
   }
+
+  /**
+   * We have observed that pex may have bugs when dealing with subject_is_issuer credentials
+   * and when handling a presentation definition with more than one input descriptor.
+   * This could be related to these issues:
+   *    - https://github.com/Sphereon-Opensource/pex/issues/96
+   *    - https://github.com/Sphereon-Opensource/pex/issues/91
+   * We are therefore trying to simplify the input to pex so remove the possibility of it generating incorrect results.
+   * Once the above issues are fixed, pex can be updated and perhaps this filtering will not needed.
+   *
+   * @param {InputDescriptorV1 | InputDescriptorV2} descriptors input descriptors
+   * @returns filtered input descriptors
+   */
+  private filterSelfSignDescriptors(
```

Comment:
@Harasz I guess your view is that `private` methods should be at the end of file? Is that what is motivating this change?

Reply:
yes, we are following that rule in other modules

Reply:
I agree with this convention. For me, this convention is coming from my experience with the C# ecosystem https://stackoverflow.com/a/310967 . Do you know where you got this convention from?

Reply:
you forced it somewhere and i liked this approach so you know :smile: